lib/model/model.go

```diff
@@ -248,10 +248,8 @@ func (m *model) StartFolder(folder string) {
 
 // Need to hold lock on m.fmut when calling this.
 func (m *model) startFolderLocked(cfg config.FolderConfiguration) {
-	if err := m.checkFolderRunningLocked(cfg.ID); err == errFolderMissing {
-		l.Warnln("Cannot start nonexistent folder", cfg.Description())
-		panic("cannot start nonexistent folder")
-	} else if err == nil {
+	_, ok := m.folderRunners[cfg.ID]
+	if ok {
```

Comment:
I think this will panic anyway, as we believe the folder is no longer in the config, so it's some sort of race.

Reply:
It shouldn't. We never rely on `m.cfg` throughout this function. As I wrote it is a race, but an expected/as design one (see https://github.com/syncthing/syncthing/issues/5298) as in first the folder was added/changed and before the model could follow through with that, it was already removed again in the config. That is not a problem for the model though, it will simply stop the folder once it gets notified about the secondary config change.